### ISSUE TITLE
Filter per Feed: Only applies to new articles (existing items in feed won't be filtered)

### ISSUE THREAD

**Issue by iyerusad** [FreshRSS/FreshRSS#2818]
Env:
---
FreshRSS 1.15.3
Hosted Docker Container

Scenario:
---
Feed: `https://www.troyhunt.com/rss/`
Filters set at feed settings:
```
intitle:'Weekly Update'
```
![image](https://user-images.githubusercontent.com/25647368/75761692-bfee5800-5cf6-11ea-82a2-c3c84a15be21.png)

Expected:
---
_Used the search bar (query: `-intitle:'Weekly Update'`) to achieve this result temporarily:_
![image](https://user-images.githubusercontent.com/25647368/75761412-46ef0080-5cf6-11ea-92bb-126c950d944f.png)

Actual:
---
![image](https://user-images.githubusercontent.com/25647368/75761512-7271eb00-5cf6-11ea-9e73-48359e2aa943.png)

---
I read documentation at https://freshrss.github.io/FreshRSS/en/users/03_Main_view.html#with-the-search-field. I assumed the filter would apply immediately/retroactively. I tried removing and adding the feed again with the filter applied.

Perhaps filtering only applies on new/incoming articles? Didn't see indication in docs, but perhaps this the case. If so, any way to trigger filter existing (other than manually setting up all queries in search bar manually and marking as read)?

**Comment by Alkarex** [comment]
Indeed, the filter action is only applied on new articles.
We could add a button to apply the filters on existing articles. Could you please update the issue title accordingly?

**Comment by iyerusad** [comment]
Yup, seeing the filters apply to incoming articles. Thank you for confirming.

A button could be nice but not required. An update to documentation could probably suffice. 
Perhaps something like:

> Filters per feed only apply to new/incoming articles - articles already fetched won't be filtered. As such, new RSS feeds won't get filtered on initial import, but future articles from the added feed will be. A filter can be tested by navigating to a feed and using the search bar.

**Comment by litecactus** [comment]
Adding in my +1 for a button to apply filters to existing articles.  Especially as needs change over time, applying filters historically is very useful.

**Comment by Alkarex** [comment]
@litecactus While waiting for such a feature, you can put the new filter in the search box, and mark all as read

**Comment by hrehfeld** [comment]
Another +1 bump :-)

**Comment by AaronKnowls** [comment]
This feature would be greatly appreciated. Specially for the ones that keep curating their feeds. 

Thank you so much for FreshRSS, it has completely change the way I use RSS. So so useful.

**Comment by Alkarex** [comment]
> Thank you so much for FreshRSS, it has completely change the way I use RSS. So so useful.

Thanks :-)

@AaronKnowls Just to be sure we talk on the same things, could you please explain your use-case more precisely?
We already have filters automatically applied to incoming articles. Is it the feature of applying to existing articles, you are asking?

**Comment by AaronKnowls** [comment]
> Is it the feature of applying to existing articles, you are asking?

Precisely!

I don't know what's wrong with my head and why this filtering system is not intuitive for me. But, that's just me. 

Thanks for your time and dedication @Alkarex.

**Comment by Frenzie** [comment]
It would be intuitive (but I'm not sure how much effort) to offer to also apply to existing articles when creating one.

**Comment by AaronKnowls** [comment]
> It would be intuitive (but I'm not sure how much effort) to offer to also apply to existing articles when creating one.

For sure, since that would be instant, you'd see immediately the effects of your filters and accelerate understanding and error fixing.
